### PR TITLE
story/io 349/admin view hashtags

### PR DESCRIPTION
**Pull request by EngbergJonas**
- new admin view
- admin view functionalities (all but hashtags disabled)
- admin hashtags view with the following functionalities:
  - display hashtags in a table
  - archive hashtags
  - add new hashtags
  - search for hashtags

### CODE REVIEW

File: src/components/Admin/AdminHashtags/AddHashtagDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          <Button onClick={onToggleAddHashtagDialog} variant="secondary">
+            {t('cancel')}
+          </Button>
+        </ActionButtons>
+      </Dialog>
+    </>
+  );
+};
+
+export default AddHashtagDialog;
```
Comment (kgunga) [inline src/components/Admin/AdminHashtags/AddHashtagDialog.tsx:95]: memoize component

File: src/components/Admin/AdminHashtags/AdminHashtagsPagination.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  page: number;
+  onPageChange: (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>, page: number) => void;
+}
+
+const AdminHashtagsPagination: FC<IAdminHashtagsPaginationProps> = ({
+  pageCount,
+  page,
+  onPageChange,
+}) => {
+  const pageHref = useCallback(() => '#', []);
```
Comment (kgunga) [inline src/components/Admin/AdminHashtags/AdminHashtagsPagination.tsx:15]: Is this supposed to be function or just value of #?

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/components/Admin/AdminHashtags/AdminHashtagsPagination.tsx (hunk 2 above)
Comment (EngbergJonas) [reply src/components/Admin/AdminHashtags/AdminHashtagsPagination.tsx:15]: This is just a memoized function that the HDS needs Pagination https://hds.hel.fi/components/pagination/code/. 

Trying to return a variable gives this error:
`Type 'string' is not assignable to type '(index: number) => string'.`

File: src/components/Admin/AdminHashtags/AddHashtagDialog.tsx (hunk 1 above)
Comment (EngbergJonas) [reply src/components/Admin/AdminHashtags/AddHashtagDialog.tsx]: fixed

File: src/components/Admin/AdminHashtags/AddHashtagDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  isOpen: boolean;
+  onToggleAddHashtagDialog: () => void;
+}
+
+const AddHashtagDialog: FC<IAddhashtagDialogProps> = ({ isOpen, onToggleAddHashtagDialog }) => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+  const { Header, Content, ActionButtons } = Dialog;
+  const [hashtagName, setHashtagName] = useState('');
+  const titleId = 'add-hashtag-title';
```
Comment (kgunga) [inline src/components/Admin/AdminHashtags/AddHashtagDialog.tsx:20]: titleId and descriptionId are immutable value so move them outside the component

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/components/Admin/AdminHashtags/AddHashtagDialog.tsx (hunk 3 above)
Comment (EngbergJonas) [reply src/components/Admin/AdminHashtags/AddHashtagDialog.tsx]: fixed

**Review by kgunga — APPROVED** [review]
👍

**Pull request merged**